### ISSUE TITLE
Digital Ocean Template Issues

### ISSUE THREAD

**Issue by bamhm182** [coder/coder#11528]
I've been working with templates a lot recently and have noticed a couple of issues with the Digital Ocean template, as well as some other things I'd like to receive some input on before attempting to fix.

### Proposed Fixes

* The metadata on [Line 275](https://github.com/coder/coder/blob/4fa07124cd7a8aee77d9bc27510e9ae4e780b7ce/examples/templates/do-linux/main.tf#L275) of the current template is attempting to check the size of the home directory mounted to `/home/coder`, but the template creates a user based on the coder username, so that mountpoint never gets created. I intend to change this to reflect the usage of the drive mounted to `/home/${data.coder_workspace.me.owner}`

### Suggestions Looking for Feedback

* DO has a droplet size with 1 CPU/512MB RAM, but it isn't available in all regions. For example, it is not available at nyc2 or nyc3, but it is available at nyc1.
* DO volumes, which are required for home drive persistence, is not available in all regions. For example, it is available at nyc1 and nyc3, but not nyc2.
* The 8 CPU/16GB RAM droplet isn't available until the account owner requests it and DO approves it.

In my personal configs, I have added the small droplet mentioned above, and removed all regions which do not support the small droplet or volumes. In my opinion, the example templates should strive to provide the most likely to succeed experience out of the box. With this in mind, I would like to propose the following changes:

* Drop the 8 CPU/16GB droplet as accounts created to test Coder on DO won't have this configuration.
* Drop regions which don't support volumes as one of the key feature of Coder is the ability to persist the home volume while destroying the vm when they're Stopped.
* Add the 1 CPU/215MB RAM droplet and remove regions that don't support it as this would decrease the cost to playing around with Coder in DO. I'm on board with having this one shot down because it seems like having more regions may be more advantageous than having a smaller/cheaper droplet.

If anyone has any feedback on these points or any other issues with the DO droplet, I'm more than happy to look at them when I make the PR for the things discussed above.

**Comment by matifali** [comment]
@bamhm182 This is a great feedback and I do agree with all proposed suggestions. We should make our example templates as general as possible so everyone can try them with minimum friction. 

We would appreciate you submitting a contribution 🙂.

**Comment by bamhm182** [comment]
I went back through and took note of which regions had which limitations. 

These regions do not support the s-1vcpu-512mb-10gb droplet:
- nyc3
- sfo2
- lon1
- tor1
- blr1

These regions do not support volumes, which are used by Coder to persist data while decreasing cost:
- nyc1
- sfo1
- ams2

All of the ones which don't support volumes have alternate data centers within the same location, but London, Toronto, and Bangalore only have 1 data center. I'm thinking of adding a note as seen below so that the regions can be included, but people will know their limitations.

![image](https://github.com/coder/coder/assets/920269/3afc79a4-58da-4801-b400-162eb3b4165d)

Reviewing my template again, I noticed another thing I noticed. Alma Linux can be created on Digital Ocean, but isn't in the template. There's also no Alma Linux SVG within the stock icons. I'm unsure about the criteria used when evaluating whether an icon should be included with stock Coder. Should I add these?
